MarsRover/Rover.cs

```diff
@@ -43,16 +91,45 @@ public Rover(Direction direction, int x, int y)
             return Direction.None;
         }
 
-        public Location Move(Command command)// TODO: does Rover need to know the Map dimensions?? // TODO: refactor
+        // Move logic:
+        // GetTargetLocation to determine where Rover should Move to
+        // use Map.GetSquareAtLocation in order to find SquareContents
+        // use Map.HasObstacle on the square to check if there is an obstacle
+        // if no obstacle, set Rover.Location to the location of that square
+        // if there is an obstacle, Rover.Location remains the same
+
+        public Location GetTargetLocation(Command command) 
         {
-            if (command == Command.Forward && Direction == Direction.North) return new Location(Location.X, Location.Y - 1);
-            if (command == Command.Forward && Direction == Direction.South) return new Location(Location.X, Location.Y + 1);
-            if (command == Command.Forward && Direction == Direction.West) return new Location(Location.X - 1, Location.Y);
-            if (command == Command.Forward && Direction == Direction.East) return new Location(Location.X + 1, Location.Y);
-            if (command == Command.Backward && Direction == Direction.North) return new Location(Location.X, Location.Y + 1);
-            if (command == Command.Backward && Direction == Direction.South) return new Location(Location.X, Location.Y - 1);
-            if (command == Command.Backward && Direction == Direction.West) return new Location(Location.X + 1, Location.Y);
-            if (command == Command.Backward && Direction == Direction.East) return new Location(Location.X - 1, Location.Y);
+            // if (command == Command.Forward && Direction == Direction.North) return new Location(Location.X, Location.Y - 1);
+            // if (command == Command.Forward && Direction == Direction.South) return new Location(Location.X, Location.Y + 1);
+            // if (command == Command.Forward && Direction == Direction.West) return new Location(Location.X - 1, Location.Y);
+            // if (command == Command.Forward && Direction == Direction.East) return new Location(Location.X + 1, Location.Y);
+            // if (command == Command.Backward && Direction == Direction.North) return new Location(Location.X, Location.Y + 1);
+            // if (command == Command.Backward && Direction == Direction.South) return new Location(Location.X, Location.Y - 1);
+            // if (command == Command.Backward && Direction == Direction.West) return new Location(Location.X + 1, Location.Y);
+            // if (command == Command.Backward && Direction == Direction.East) return new Location(Location.X - 1, Location.Y);
+            if (command == Command.Forward) // TODO: variables for Location.X & Location.Y et al??

```

Comment:
can the forward and the backward blocks be extracted to their own functions to look like the `Execute` method above?

Reply:
I think your switch statements look fine as it is (no need for variables)